lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -660,6 +661,62 @@ int amf_app_handle_pdu_session_response(
 
   return rc;
 }
+/****************************************************************************
+ **                                                                        **
+ ** Name:    convert_ambr()                                                **
+ **                                                                        **
+ ** Description: Converts the session ambr format from                     **
+ **  one defined in create_pdu_session_response to one defined in          **
+ **  pdu_session_estab_accept message                                      **
+ **                                                                        **
+ ** Inputs:  pdu_ambr_response_unit, pdu_ambr_response_value               **
+ **          ambr_unit, ambr_value                                         **
+ **                                                                        **
+ **      Return:   void                                                    **
+ **                                                                        **
+ ***************************************************************************/
+void convert_ambr(
+    uint32_t* pdu_ambr_response_unit, uint32_t* pdu_ambr_response_value,
+    uint8_t* ambr_unit, uint16_t* ambr_value) {
+  int count = 1;
+  if (*pdu_ambr_response_unit == KBPS) {
+    *pdu_ambr_response_value *= 1000;
+  }
+  uint32_t temp_pdu_ambr_response_value = *pdu_ambr_response_value;
+  if (temp_pdu_ambr_response_value / 1000 == 0) {
+    count = 1;
+  } else {
+    do {
+      temp_pdu_ambr_response_value = (temp_pdu_ambr_response_value / 1000);
+      count++;
+    } while (temp_pdu_ambr_response_value >= AMBR_UNIT_CONVERT_THRESHOLD);
+  }
+
+  switch (count) {
+    case 1:
+      *ambr_unit =
+          static_cast<uint8_t>(magma5g::M5GSessionAmbrUnit::VALUE_NOT_USED);
+      break;
+    case 2:
+      *ambr_unit = static_cast<uint8_t>(
+          magma5g::M5GSessionAmbrUnit::MULTIPLES_1KBPS);  // Kbps
+      break;
+    case 3:
+      *ambr_unit = static_cast<uint8_t>(
+          magma5g::M5GSessionAmbrUnit::MULTIPLES_1MBPS);  // Mbps
+      break;
+    case 4:
+      *ambr_unit = static_cast<uint8_t>(
+          magma5g::M5GSessionAmbrUnit::MULTIPLES_1GBPS);  // Gbps
+      break;
+    default:
+      *ambr_unit = static_cast<uint8_t>(
+          magma5g::M5GSessionAmbrUnit::MULTIPLES_1GBPS);  // Gbps
+      break;
+  }
+  int k       = pow(1000, (count - 1));
+  *ambr_value = static_cast<uint16_t>(*pdu_ambr_response_value / k);
```

Comment:
you should be able to get the value from `temp_pdu_ambr_response_value`.

Reply:
Comment addressed.

Reply:
Comment addressed.

Reply:
Comment addressed.